robottelo/host_helpers/repository_mixins.py

```diff
@@ -0,0 +1,743 @@
+            {
+                'organization-id': org_id,
+                'lifecycle-environment-id': lce_id,
+                'content-view-id': content_view_id,
+            }
+        )
+        # Add subscriptions to activation-key
+        # Get organization subscriptions
+        subscriptions = self.satellite.cli.Subscription.list(
+            {'organization-id': org_id}, per_page=False
+        )
+        added_subscription_names = []
+        for subscription in subscriptions:
+            if (
+                subscription['name'] in subscription_names
+                and subscription['name'] not in added_subscription_names
+            ):
+                self.satellite.cli.ActivationKey.add_subscription(
+                    {
+                        'id': activation_key['id'],
+                        'subscription-id': subscription['id'],
+                        'quantity': 1,
+                    }
+                )
+                added_subscription_names.append(subscription['name'])
+                if len(added_subscription_names) == len(subscription_names):
+                    break
+        missing_subscription_names = set(subscription_names).difference(
+            set(added_subscription_names)
+        )
+        if missing_subscription_names:
+            raise ValueError(f'Missing subscriptions: {missing_subscription_names}')
+        return activation_key
+
+    def organization_has_manifest(self, organization_id):
+        """Check if an organization has a manifest, an organization has manifest if one of it's
+        subscriptions have the account defined.
+        """
+        subscriptions = self.satellite.cli.Subscription.list(
+            {'organization-id': organization_id}, per_page=False
+        )
+        return any(bool(sub['account']) for sub in subscriptions)
+
+    def setup_content(
+        self,
+        org_id,
+        lce_id,
+        upload_manifest=False,
+        download_policy=constants.DOWNLOAD_POLICY_ON_DEMAND,
+        rh_subscriptions=None,
+    ):
+        """
+        Setup content view and activation key of all the repositories.
+
+        :param org_id: The organization id
+        :param lce_id:  The lifecycle environment id
+        :param upload_manifest: Whether to upload the manifest (The manifest is
+            uploaded only if needed)
+        :param download_policy: The repositories download policy
+        :param rh_subscriptions: The RH subscriptions to be added to activation
+            key
+        """
+        if self._repos_info:
+            raise RepositoryAlreadyCreated('Repositories already created can not setup content')
+        if rh_subscriptions is None:
+            rh_subscriptions = []
+        if self.need_subscription:
+            # upload manifest only when needed
+            if upload_manifest and not self.organization_has_manifest(org_id):
+                manifests.upload_manifest_locked(org_id, interface=manifests.INTERFACE_CLI)
```

Comment:
Good Catch!  I think we need to get the manifest helpers refactor get in first!

Reply:
BTW, This could be a problem with destructive_sat marked tests only and in this entire refactor there is only one destructive test `test_content_access_after_stopped_foreman` using manifest upload. So considering that test being failed until we do manifest helpers refactor.  There are many destructive tests though that needs to recover from `entities.` behaviour, all pointing to default satellite.

So lets not block this PR !!